be/src/io/input_stream.h

```diff
@@ -0,0 +1,33 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2022-present, StarRocks Limited.
+
+#pragma once
+
+#include <string_view>
+
+#include "io/readable.h"
+
+namespace starrocks::io {
+
+class InputStream : public Readable {
```

Comment:
Would you consider support any **Buffering** semantic to the `Stream` interface ? Read buffer or write buffer?

Reply:
> 

You can check the `InputStream::peak()` and `OutputStream::get_direct_buffer()` and see if they can saitisfy the buffering semantic you need.
BTW, I plan to add a utility class `BufferedInputStream` and `BufferedOutputStream` later:
```
class BufferedInputStream : public InputStream {
public:
    BufferedInputStream(const void* buff, int64_t size, InputStream* raw_stream);
};

class BufferedOutputStream : public OutputStream {
public:
    BufferedOutputStream(void* buff, int64_t size, OutputStream* raw_stream);
};
```

Reply:
Could you please add some explanation for all classes you added.
It will help people to know them through the comment.
These base interfaces should be declared in more details, such as the differences between each other.

For example except Stream is a type of Readable, is there any other types of Readable. And what's the differences.

This can help others to know why you create this interface other than another one.

Reply:
> Could you please add some explanation for all classes you added. It will help people to know them through the comment. These base interfaces should be declared in more details, such as the differences between each other.
> 
> For example except Stream is a type of Readable, is there any other types of Readable. And what's the differences.
> 
> This can help others to know why you create this interface other than another one.

I added some simple class comments